go-controller/pkg/config/cni.go

```diff
@@ -26,6 +26,7 @@ func WriteCNIConfig() error {
 		LogFileMaxSize:    Logging.LogFileMaxSize,
 		LogFileMaxBackups: Logging.LogFileMaxBackups,
 		LogFileMaxAge:     Logging.LogFileMaxAge,
+		Kubeconfig:        Kubernetes.Kubeconfig,
```

Comment:
Is this even set for containerized deployments? We could easily use in-cluster config.

Reply:
Likewise, it may not be the same path on disk inside the container.

Reply:
@squeed this is actually for "unprivileged" mode where the CNI plugin (run by CRIO/containerd on the host) is responsible for setting up the pod networking after ovnkube-node returns the IPAM info. Girish uses that mode, but OpenShift does not. So that's already uncontainerized. "privileged mode" is what we currently use in OpenShift and that is containerized, and this option is not used there.

Reply:
@girishmg what's missing is a good way to get the kubeconfig (or otherwise create a kubeclient) in unprivileged mode... since we may not have a kubeconfig inside the ovnkube-node container that we could copy to a location the cni shim can read it?

Reply:
@girishmg @squeed I made ovnkube-node pass either the kubeconfig or the URL/Token/CA via the CNI config file. That should allow the InClusterConfig() info to be used by the shim in unprivileged mode if we aren't using a kubeconfig (which most of the time we aren't). Any security implications you can think of there? I hope we don't let random containers map /etc/cni/networks and mess with stuff... :)